internal/controller/dnsrecord_controller.go

```diff
@@ -156,15 +156,25 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 }
 
 func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, current *v1alpha1.DNSRecord, requeueAfter time.Duration) (reconcile.Result, error) {
-	current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+	if requeueAfter != noRequeueDuration {
+		current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+	}
```

Comment:
This is one of two problems this PR is solving - constant updates to this field were causing the reconciler to go nuts. If we aren't setting/maintaining this field always (in other words it won't always be a reliable source of information) we better reset it to zero time. With something like this:
```
if requeueAfter != noRequeueDuration {
		current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
	} else {
		current.Status.QueuedFor = metav1.NewTime(time.Time{})
	}
```
wdyt?